src/erasure-code/clay/ErasureCodeClay.cc

```diff
@@ -388,7 +388,9 @@ void ErasureCodeClay::get_repair_subchunks(const int &lost_node,
 
 int ErasureCodeClay::get_repair_sub_chunk_count(const set<int> &want_to_read)
 {
-  int weight_vector[t] = {0};
+  int weight_vector[t];
+  weight_vector = {0};
```

Comment:
this still does not work. we still have
```
error: variable-sized object may not be initialized
```

the crux is that `t` is determined at run-time. so probably we can fix it like
```c++
std::fill(weight_vector, weight_vector + t, 0);
```

Reply:
@tchaikov 
Right, ... Funny. First time that GCC is more picky than Clang for this.
Tried to avoid memset and likes, but I guess std:fill() it is going to be.